geo-types/src/geometry/multi_line_string.rs

```diff
@@ -118,6 +120,36 @@ impl<T: CoordNum> MultiLineString<T> {
     }
 }
 
+#[cfg(feature = "multithreading")]
+impl<T: CoordNum + Send> IntoParallelIterator for MultiLineString<T> {
+    type Item = LineString<T>;
+    type Iter = rayon::vec::IntoIter<LineString<T>>;
+
+    fn into_par_iter(self) -> Self::Iter {
+        self.0.into_par_iter()
+    }
+}
+
+#[cfg(feature = "multithreading")]
```

Comment:
The consuming `into` implementation makes sense to me.

And I can see how the ref and mut flavors below provide necessary **non**-consuming functionality. But what does it look like to use it? Can you include some example code in your PR that uses these new methods to make the new interface easier to evaluate?

My guess is that you'd need to do something like

`(& my_multi_line_string).into_par_iter()`
and
`(&mut my_multi_line_string).into_par_iter()`
?

Which is a bit rough. I'm afraid people will just end up `my_multi_line_string.clone().into_par_iter()` out of confusion.

Would it get better if we got rid of these two ref/mut implementations and instead implemented `rayon::IntoParallelRefIterator` and `rayon::IntoParallelRefMutIterator` which seem purpose built for this?

Reply:
Hm. These work with the current impls:


```rust
    #[test]
    fn test_multithreading_linestring() {
        let multi: MultiLineString<i32> = wkt! {
            MULTILINESTRING((0 0,2 0,1 2,0 0), (10 10,12 10,11 12,10 10))
        };
        let mut multimut: MultiLineString<i32> = wkt! {
            MULTILINESTRING((0 0,2 0,1 2,0 0), (10 10,12 10,11 12,10 10))
        };
        let _ = multi.par_iter().for_each(|_p| ());
        let _ = multimut.par_iter_mut().for_each(|_p| ());
        let _ = &multi.into_par_iter().for_each(|_p| ());
        let _ = &mut multimut.par_iter_mut().for_each(|_p| ());
    }
```

I admittedly didn't see the `rayon::IntoParallelRefIterator` and just adapted our existing iterators to `rayon::IntoParallelIterator`, but I'm confused by

> This trait is automatically implemented for I where &I: IntoParallelIterator. In most cases, users will want to implement [IntoParallelIterator](https://docs.rs/rayon/latest/rayon/iter/trait.IntoParallelIterator.html) rather than implement this trait directly.

Reply:
Oh splendid! They're already implemented by the blanket impl!

I'm glad that users will be able to do:
```
let _ = multi.par_iter().for_each(|_p| ());
let _ = multimut.par_iter_mut().for_each(|_p| ());
```

vs. the slightly more arcane:
```
let _ = &multi.into_par_iter().for_each(|_p| ());
let _ = &mut multimut.into_par_iter().for_each(|_p| ());
```

Reply:
Oh yeah the latter two were just to make sure "everything" worked. Thanks for the sanity check!